bitcoin/tx.c

```diff
@@ -98,28 +98,43 @@ static struct amount_sat bitcoin_tx_compute_fee(const struct bitcoin_tx *tx)
 	return fee;
 }
 
-int elements_tx_add_fee_output(struct bitcoin_tx *tx)
+/**
+ * Add an explicit fee output if necessary.
+ *
+ * An explicit fee output is only necessary if we are using an elements
+ * transaction, and we have a non-zero fee. This method may be called multiple
+ * times.
+ *
+ * Returns the position of the fee output, or -1 in the case of non-elements
+ * transactions.
+ */
+static int elements_tx_add_fee_output(struct bitcoin_tx *tx)
 {
 	struct amount_sat fee = bitcoin_tx_compute_fee(tx);
-	int pos = -1;
+	int pos;
 	struct witscript *w;
 
 	/* If we aren't using elements, we don't add explicit fee outputs */
 	if (!chainparams->is_elements || amount_sat_eq(fee, AMOUNT_SAT(0)))
 		return -1;
 
 	/* Try to find any existing fee output */
-	for (int i=0; i<tx->wtx->num_outputs; i++) {
-		if (elements_tx_output_is_fee(tx, i)) {
-			assert(pos == -1);
-			pos = i;
-		}
+	for (pos = 0; pos < tx->wtx->num_outputs; pos++) {
+		if (elements_tx_output_is_fee(tx, pos))
+			break;
 	}
 
-	if (pos == -1) {
+	if (pos >= tx->wtx->num_outputs) {
```

Comment:
It cannot be `>`, can it ?

Reply:
You're right. For some reason I'm in the habit of being overly cautious in these cases (spurious increments somewhere), but I'll fixup in this case :-)